cmd/controller/app/options/options.go

```diff
@@ -200,6 +205,10 @@ func (s *ControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.DefaultACMEIssuerDNS01ProviderName, "default-acme-issuer-dns01-provider-name", defaultACMEIssuerDNS01ProviderName, ""+
 		"Required if --default-acme-issuer-challenge-type is set to dns01. The DNS01 provider to use for ingresses using ACME dns01 "+
 		"validation that do not explicitly state a dns provider.")
+	fs.BoolVar(&s.DNS01CheckAuthoritative, "dns01-check-authoritative", defaultDNS01CheckAuthoritative,
+		"Whether the authoritative name server for a resource record is checked or"+
```

Comment:
Could you expand out this flag description a bit more? The comment on the type provides a bit more insight than this description, but the type comment will be a lot harder for users to find imo:

```
// DNS01CheckAuthoritative is a flag for controlling if auth nss are used
// for checking propogation of an RR. This is the ideal scenario
```

(with a bit of cleaning up in order to have it as a flag!)

Reply:
nit: propogation -> propagation

Reply:
Done & Done "I think"